PHP Span Metrics (#11031)
getsentry/sentry-docs: docs/platforms/php/common/metrics/index.mdx

---
title: Set Up Metrics
description: "Learn how to measure the data points you care about by configuring Metrics in your PHP app."
sidebar_order: 5000
---

<Include name="feature-stage-beta-metrics.mdx" />

Sentry metrics are numerical values that allow you to pinpoint and solve issues impacting user experience and app performance. You can measure and track the data points that are important to you and your application, such as latency, processing time, event size or number of user signups. This gives you a broad overview of the behaviour and health of your system.

> In Sentry, metrics are extracted from your span data, to ensure they are contextually linked to the rest of your data through Tracing, so you can do more than just observe. If you notice an unexpected pattern, you can zoom in to see a snapshot through your entire system with a trace and get rich additional debugging information through issues, replays or profiles to help you understand the root cause and take action.

Reviewer: ```suggestion
In Sentry, metrics are extracted from your span data, to ensure they are contextually linked to the rest of your data through [tracing](/platforms/php/tracing/), so you can do more than just observe. If you notice an unexpected pattern, you can zoom in to see a snapshot through your entire system with a trace and get rich additional debugging information through issues, replays or profiles to help you understand the root cause and take action.
```

## How to attach data to a span

> Sentry's tracing product has to be enabled in order for Metrics to work. To begin capturing metrics data, you first need to start a transaction. Check out the <PlatformLink to="/tracing/">performance setup</PlatformLink> and <PlatformLink to="/tracing/instrumentation/custom-instrumentation/">custom instrumentation</PlatformLink> documentation for more detailed information.

Reviewer: ```suggestion
Sentry's tracing product has to be enabled for Metrics to work. Once this is done, you'll need to start a transaction to begin capturing metrics data. Check out the <PlatformLink to="/tracing/">performance setup</PlatformLink> and <PlatformLink to="/tracing/instrumentation/custom-instrumentation/">custom instrumentation</PlatformLink> documentation for more detailed information.
```

```php
// Fetch the current span
$span = \Sentry\SentrySdk::getCurrentHub()->getSpan();
if ($span !== null) {
    // Attach the data to the span
    $span->setData([
        'foo' => 'bar',
        'baz' => 123,
    ]);
}
```

> ## Further examples on how to attach certain measurements to a span

Reviewer: ```suggestion
## More Examples 

Here's another example showing how to attach certain measurements to a span:
```

```php
$span = \Sentry\SentrySdk::getCurrentHub()->getSpan();
if ($span !== null) {
    $span->setData([
        'processing_time' => $event->getProcessingTime(),
        'user_email' => $identity->get('email'),
        'memory_peak_usage' => memory_get_peak_usage(false),
    ]);
}
```

> After you emit your span data, you need to configure the metric you want to track in the Sentry UI from this data field. Learn more about this [here](/product/explore/metrics/metrics-set-up/).

Reviewer: ```suggestion
Once your span data's been emitted, you'll need to configure the metric you want to track in the Sentry UI from this data field. Learn more about [tracking specific data points](/product/explore/metrics/).
```
